### PR TITLE
CDEV-204 patient history empty states

### PR DESCRIPTION
**Pull request by sarahmangels**
expand table empty states to include request records button and show spinner while request is in progress

https://zeushealth.atlassian.net/browse/CDEV-204

image

image

### CODE REVIEW

File: src/components/content/patient-history/use-patient-history.tsx [hunk 1]
```diff
@@ -172,3 +177,51 @@ export function usePatientHistoryDetails() {
...
+      if (patient.data && !isPolling) {
+        setIsPolling(true);
+        const requestContext = await getRequestContext();
+        await pollPatientHistory(requestContext, patient.data.id);
+        setIsFetching(false);
+      }
+    }
+    void poll();
+  }, [getRequestContext, patient, isPolling]);
+  return isFetching;
```
Comment (eric-gunther) [inline src/components/content/patient-history/use-patient-history.tsx:197]: Should there be more uses of `setIsPolling` and `setIsFetching` here?

File: src/components/content/patient-history/use-patient-history.tsx [hunk 2]
```diff
@@ -172,3 +177,51 @@ export function usePatientHistoryDetails() {
...
+      const response = (await getBuilderRefreshHistoryMessages({
+        requestContext,
+        patientId,
+      })) as PatientHistoryJobResponse;
+
+      isDone = response.data[0]?.attributes.providers.every(
+        (provider) =>
+          provider.status === "done" ||
+          provider.status === "error" ||
+          provider.status === "done_with_errors"
```
Comment (eric-gunther) [inline src/components/content/patient-history/use-patient-history.tsx:214]: We reference these statuses in a number of places now (eg. `patient-history-message-status.tsx`, `patient-history-table.tsx`), think we should have em as constants.

File: src/components/core/icons/refresh.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+export const RefreshIcon = ({ className }: { className?: string }) => (
```
Comment (eric-gunther) [inline src/components/core/icons/refresh.tsx:1]: Is there not an applicable font awesome icon for this?

**Review by eric-gunther — CHANGES_REQUESTED** [review]

File: src/components/core/icons/refresh.tsx (hunk 3 above)
Comment (sarahmangels) [reply src/components/core/icons/refresh.tsx:1]: no :( Tai said the closest one in font awesome looked too much like an undo icon so she modified it

File: src/components/content/patient-history/use-patient-history.tsx (hunk 1 above)
Comment (sarahmangels) [reply src/components/content/patient-history/use-patient-history.tsx:197]: setIsPolling is being used to block this hook from being executed multiple times at once - so there can only be 1 poll going on at a time, so it makes sense to only call it once. 

setIsFetching is used to mark when the polling is complete so we only set it at the end once the API says the job is complete!

File: src/components/content/patient-history/use-patient-history.tsx (hunk 2 above)
Comment (sarahmangels) [reply src/components/content/patient-history/use-patient-history.tsx:214]: hm wait i think it sort of is already?? its comparing provider.status which is type PatientRefreshHistoryMessageStatus (https://github.com/zeus-health/ctw-component-library/blob/5d8b23156ed5e9daf5bcf88867f4ef903e5dcd63/src/services/patient-history/patient-history-types.ts#L1) so it wont let me change the strings here to anything outside of that type

Comment (eric-gunther) [reply src/components/content/patient-history/use-patient-history.tsx:214]: Ah, well that seems fair enough. Won't block on it.

**Review by eric-gunther — DISMISSED** [review]

**Review by eric-gunther — APPROVED** [review]

**Pull request merged**